### PR TITLE
Updated default and hover state styling - RSC-679

### PR DESCRIPTION
**Pull request by entroform**
https://issues.sonatype.org/browse/RSC-679

In addition to NxCheckbox and NxRadio changes, I updated `--nx-color-form-element-border` CSS variable with `--nx-swatch-grey-50` to match Kaleido. This might will change the border color for other form elements as side effect.

### CODE REVIEW

**Comment by rpokorny** [comment]
> This might will change the border color for other form elements as side effect.

You should determine which ones change and whether those changes match kaleido.  One case I see is that this PR changes the hover border on `NxColorPicker` in a way that does not match the spec for that component.  That should be addressed here.

**Comment by entroform** [comment]
> > This might will change the border color for other form elements as side effect.
> 
> You should determine which ones change and whether those changes match kaleido. One case I see is that this PR changes the hover border on `NxColorPicker` in a way that does not match the spec for that component. That should be addressed here.

Thanks, going to look into it and address it. Should I update the ticket to reflect changes affected by `--nx-color-form-element-border`?

**Comment by rpokorny** [comment]
> Thanks, going to look into it and address it. Should I update the ticket to reflect changes affected by --nx-color-form-element-border?

The original purpose of the ticket is what it is (and shouldn't be changed at this point IMO) but if other "bugs" (mismatches with kaleido) get fixed in the process of fixing this one it wouldn't hurt to mention them in a comment on the JIRA ticket

**Comment by glenhunter** [comment]
I think that `NxRadio` needs the same border adjustment(s)

**Comment by entroform** [comment]
> > Thanks, going to look into it and address it. Should I update the ticket to reflect changes affected by --nx-color-form-element-border?
> 
> The original purpose of the ticket is what it is (and shouldn't be changed at this point IMO) but if other "bugs" (mismatches with kaleido) get fixed in the process of fixing this one it wouldn't hurt to mention them in a comment on the JIRA ticket

Fixed other mismatches affected by  `--nx-color-form-element-border` here :)
6b7d61614e64372ad4c8eab6bb09f948817ed8b3

I also noticed that the active state for radios and checkboxes also do no align; I updated it in the above commit.
Let me know if you find any issues with the implementation 👍

File: lib/src/components/NxCheckbox/NxCheckbox.scss [hunk 1]
```diff
@@ -32,11 +32,22 @@ $-width: 16px;
...
+    visibility: visible;
+  }
+}
+
 .nx-checkbox__input:checked:disabled {
   + .nx-checkbox__box {
     background-color:var(--nx-color-interactive-background-disabled);
     border-color: var(--nx-color-interactive-border-disabled);
-
+
```
Comment (glenhunter) [inline lib/src/components/NxCheckbox/NxCheckbox.scss:50]:
```suggestion

```

File: lib/src/components/NxCheckbox/NxCheckbox.tsx [hunk 2]
```diff
@@ -56,7 +56,7 @@ const NxCheckbox = forwardRef<HTMLLabelElement, Props>(
             {/* Put a space in the box if not checked,
               * in order to provide a consistent vertical-align baseline
               */}
-            { isChecked ? <FontAwesomeIcon icon={faCheck} /> : '\u00A0' }
+            <FontAwesomeIcon icon={faCheck} />
```
Comment (glenhunter) [inline lib/src/components/NxCheckbox/NxCheckbox.tsx:59]: Two things: is this in scope? if so do we still need the comment above?

Comment (entroform) [reply lib/src/components/NxCheckbox/NxCheckbox.tsx:59]: > Two things: is this in scope? if so do we still need the comment above?

Hi Glen, I believe so. This change is needed because the icon needs to be visible in the `active` state as well as the`checked` state according to Kaleido. Nice catch, I forgot to remove the comment.

Comment (glenhunter) [reply lib/src/components/NxCheckbox/NxCheckbox.tsx:56]: > > Two things: is this in scope? if so do we still need the comment above?
> 
> Hi Glen, I believe so. This change is needed because the icon needs to be visible in the `active` state as well as the`checked` state according to Kaleido. Nice catch, I forgot to remove the comment.

Well technically the `:active` state of a checkbox (does a checkbox even have an `active` state?) could be either of the state of checking it or unchecking it. The checkmark icon should only appear when the checkbox is in the `:checked` state

**Comment by entroform** [comment]
I started a conversation about the active state on checkboxes and radios in the #react-components channel.
It looks like @alexalyons126 will be updating it in Kaleido and once that's done I'll update this PR 👍.

File: lib/src/components/NxCheckbox/NxCheckbox.tsx (hunk 2 above)
Comment (entroform) [reply lib/src/components/NxCheckbox/NxCheckbox.tsx:56]: > > > Two things: is this in scope? if so do we still need the comment above?
> > 
> > 
> > Hi Glen, I believe so. This change is needed because the icon needs to be visible in the `active` state as well as the`checked` state according to Kaleido. Nice catch, I forgot to remove the comment.
> 
> Well technically the `:active` state of a checkbox (does a checkbox even have an `active` state?) could be either of the state of checking it or unchecking it. The checkmark icon should only appear when the checkbox is in the `:checked` state

Yeah, I agree that it should only appear in the `:checked` state. I was confused because in the Kaleido the `:active` state is the same as the `:checked` state. I started a conversation about it in the #react-components channel and it looks like @alexalyons126 will be updating it

Comment (entroform) [reply lib/src/components/NxCheckbox/NxCheckbox.tsx]: > > > > Two things: is this in scope? if so do we still need the comment above?
> > > 
> > > 
> > > Hi Glen, I believe so. This change is needed because the icon needs to be visible in the `active` state as well as the`checked` state according to Kaleido. Nice catch, I forgot to remove the comment.
> > 
> > 
> > Well technically the `:active` state of a checkbox (does a checkbox even have an `active` state?) could be either of the state of checking it or unchecking it. The checkmark icon should only appear when the checkbox is in the `:checked` state
> 
> Yeah, I agree that it should only appear in the `:checked` state. I was confused because in the Kaleido the `:active` state is the same as the `:checked` state. I started a conversation about it in the #react-components channel and it looks like @alexalyons126 will be updating it

After offline discussions, the decision was to revert back to how it was before. 44c1e9043dad5474c028b5d79b166e6b7e7baaef

File: lib/src/base-styles/_nx-radio-checkbox.scss [hunk 3]
```diff
@@ -48,7 +48,7 @@
 
     &:hover {
       .nx-radio-checkbox__control {
-        border-color: var(--nx-color-form-element-border);
+        border-color: var(--nx-swatch-grey-05);
         filter: none;
         stroke: var(--nx-swatch-grey-05);
```
Comment (rpokorny) [inline lib/src/base-styles/_nx-radio-checkbox.scss:53]: I don't think border and stroke are actually necessary here at all; they just repeat what's already present via the non-focus `:hover` selector

File: lib/src/components/NxCheckbox/NxCheckbox.scss [hunk 4]
```diff
@@ -36,7 +36,6 @@ $-width: 16px;
   + .nx-checkbox__box {
     background-color:var(--nx-color-interactive-background-disabled);
     border-color: var(--nx-color-interactive-border-disabled);
-
```
Comment (rpokorny) [inline lib/src/components/NxCheckbox/NxCheckbox.scss:39]: Extraneous and undesirable change

File: lib/src/components/NxCodeSnippet/NxCodeSnippet.scss [hunk 5]
```diff
@@ -44,7 +44,7 @@
     width: 100%;
 
     .nx-text-input__box {
-      border-color: var(--nx-color-form-element-border);
+      border-color: var(--nx-swatch-grey-05);
```
Comment (rpokorny) [inline lib/src/components/NxCodeSnippet/NxCodeSnippet.scss:47]: @alexalyons126 is this actually what we want? It matches kaleido but effectively what this means is that the text field in an `NxCodeSnippet` would always look the way a _hovered_ text input is supposed to.

File: lib/src/base-styles/_nx-radio-checkbox.scss (hunk 3 above)
Comment (entroform) [reply lib/src/base-styles/_nx-radio-checkbox.scss]: > I don't think border and stroke are actually necessary here at all; they just repeat what's already present via the non-focus `:hover` selector

Thanks! Removed here:
5c5ea6780bb688265db64fa897b7852cd8c5935e

File: lib/src/components/NxCheckbox/NxCheckbox.scss (hunk 4 above)
Comment (entroform) [reply lib/src/components/NxCheckbox/NxCheckbox.scss]: > Extraneous and undesirable change

Ops, totally missed this. Fixed:
5c5ea6780bb688265db64fa897b7852cd8c5935e

File: lib/src/components/NxCodeSnippet/NxCodeSnippet.scss (hunk 5 above)
Comment (entroform) [reply lib/src/components/NxCodeSnippet/NxCodeSnippet.scss]: > > @alexalyons126 is this actually what we want? It matches kaleido but effectively what this means is that the text field in an `NxCodeSnippet` would always look the way a _hovered_ text input is supposed to.
> 
> I reached out to @alexalyons126 on Slack to confirm.

Confirmed with @alexalyons126 that this should match other form fields.
Reverted here:
4c97190c807354f3883f3f2691fe6a38a1f3c2e7

**Review by rpokorny — APPROVED** [review]
